store/pg_engine.go

```diff
@@ -227,91 +217,146 @@ func (db *DB) seedFile(name string) error {
+func (db *DB) migrate(ctx context.Context, d dbdriver.Driver, curVer *semver.Version, databaseName string) error {
 	db.l.Info("Apply database migration if needed...")
-	db.l.Info(fmt.Sprintf("Current schema version before migration: %s", curVer))
-
-	// major version is 0 when the store isn't yet setup for the first time.
-	if curVer.major != 0 && curVer.major != majorSchemaVervion {
-		return fmt.Errorf("current major schema version %d is different from the major schema version %d this release %s expects", curVer.major, majorSchemaVervion, db.serverVersion)
+	if curVer == nil {
+		db.l.Info("The database schema has not been setup.")
+	} else {
+		db.l.Info(fmt.Sprintf("Current schema version before migration: %s", curVer))
+		major := (*curVer).Major
+		if major != majorSchemaVervion {
+			return fmt.Errorf("current major schema version %d is different from the major schema version %d this release %s expects", major, majorSchemaVervion, db.serverVersion)
+		}
 	}
 
-	if curVer.major == 0 && curVer.minor == 0 {
-		createDatabaseStatement := fmt.Sprintf("CREATE DATABASE %s", databaseName)
-		// TODO(d): support semantic versioning.
+	// Initial schema setup.
+	if curVer == nil {
+		latestSchemaPath := fmt.Sprintf("migration/%s/%s", db.schemaVersion.String(), latestSchemaFile)
+		buf, err := migrationFS.ReadFile(latestSchemaPath)
+		if err != nil {
+			return fmt.Errorf("failed to read latest schema %q, error %w", latestSchemaPath, err)
+		}
+		latestDataSchemaPath := fmt.Sprintf("migration/%s/%s", db.schemaVersion.String(), latestDataFile)
+		dataBuf, err := migrationFS.ReadFile(latestDataSchemaPath)
+		if err != nil {
+			return fmt.Errorf("failed to read latest schema %q, error %w", latestSchemaPath, err)
+		}
+		// We will create the database together with initial schema and data migration.
+		stmt := fmt.Sprintf("CREATE DATABASE %s;\n\\connect \"%s\";\n%s\n%s", databaseName, databaseName, buf, dataBuf)
 		if _, _, err := d.ExecuteMigration(
 			ctx,
 			&dbdriver.MigrationInfo{
-				ReleaseVersion: db.serverVersion,
-				Version:        createDatabaseSchemaVersion,
-				Namespace:      databaseName,
-				Database:       databaseName,
-				Environment:    "", /* unused in execute migration */
-				Source:         dbdriver.LIBRARY,
-				Type:           dbdriver.Baseline,
-				Description:    fmt.Sprintf("Create database %s.", databaseName),
-				CreateDatabase: true,
+				ReleaseVersion:        db.serverVersion,
+				UseSemanticVersion:    true,
+				Version:               db.schemaVersion.String(),
+				SemanticVersionSuffix: time.Now().Format("20060102150405"),
+				Namespace:             databaseName,
+				Database:              databaseName,
+				Environment:           "", /* unused in execute migration */
+				Source:                dbdriver.LIBRARY,
+				Type:                  dbdriver.Baseline,
+				Description:           fmt.Sprintf("Migrate %s.", latestSchemaPath),
```

Comment:
Provide a more readable description such as including the server version, schema version and we are init the schema

Reply:
Done.